Migration guide to dbt utils v1 (#2471)
fishtown-analytics/docs.getdbt.com: website/docs/guides/migration/versions/04-upgrading-to-dbt-utils-v1.0.md

---

> title: "Migrating to dbt utils 1.0"

Reviewer: Adding the below to be consistent with the sidebar, but wanted to double check with you if 'upgrading' is the correct term? or is it 'migrating'? Also, should we use v1.0 so it reads like the rest of the migration guides?

```suggestion
title: "Upgrading to dbt utils 1.0"
```

---

> # dbt utils 1.0 Migration Guide

Reviewer: editing in case we do want to match with the rest of the guides

```suggestion
# Upgrading to dbt utils 1.0
```

For the first time, dbt utils is crossing the major version boundary. From [last month’s blog post](https://www.getdbt.com/blog/announcing-dbt-v1.3-and-utils/): 

> > It’s time to formalise what was already unofficial policy: you can rely on dbt utils in the same way as you do dbt Core, with stable interfaces and consistent and intuitive naming.

Reviewer: @joellabes it looks like the quote is different from what's in the blog - do we want to change the below so it matches with the blog?

```suggestion
>It’s time for dbt utils to tick over to 1.0 and commit to what was already unofficial policy: you can rely on dbt utils in the same way as you do dbt Core, with stable interfaces and consistent and intuitive naming.
```

Author: I took some liberties in quoting myself, so that it read better in isolation. I think it's OK to stay as-is

---
title: "Upgrading to dbt utils v1.0"
---

# Upgrading to dbt utils v1.0

For the first time, [dbt utils](https://hub.getdbt.com/dbt-labs/dbt_utils/latest/) is crossing the major version boundary. From [last month’s blog post](https://www.getdbt.com/blog/announcing-dbt-v1.3-and-utils/): 

> > It’s time to formalise what was already unofficial policy: you can rely on dbt utils in the same way as you do dbt Core, with stable interfaces and consistent and intuitive naming.

Reviewer: changing this to us spelling

```suggestion
> It’s time to formalize what was already unofficial policy: you can rely on dbt utils in the same way as you do dbt Core, with stable interfaces and consistent and intuitive naming.
```

> Just like the switch to dbt Core 1.0 last year, there are some breaking changes as we standardized and prepared for the future. Most changes can be handled with find-and-replace. If you need help, post on the [Community Forum](https://discourse.getdbt.com) or in #package-ecosystem on Slack. 

Reviewer: ```suggestion
Just like the switch to dbt Core 1.0 last year, there are some breaking changes as we standardized and prepared for the future. Most changes can be handled with find-and-replace. If you need help, post on the [Community Forum](https://discourse.getdbt.com) or in [#package-ecosystem](https://getdbt.slack.com/archives/CU4MRJ7QB) channel on Slack. 
```

- Several long-promised deprecations to `deduplicate()` have been applied:
    - The `group_by` argument is replaced by `partition_by`.
    - `relation_alias` is removed. If you need an alias, you can pass it directly to the `relation` argument.
    - `order_by` is now mandatory. Pass a static value like `1` if you don’t care how they are deduplicated.
- The deprecated `table` argument has been removed from `unpivot()`. Use `relation` instead.

## Resolving error messages
After upgrading, these are common error messages you may encounter, along with their resolutions.
<details>
	<summary><code>dict object has no attribute MACRO_NAME</code></summary>
	<div>
		<p><b>Cause</b>: No macro called <code>MACRO_NAME</code> exists. This is most likely because the macro has moved to the <code>dbt</code> namespace (see above). It could also be because you haven't run dbt deps or have misspelled a macro's name.</p>
		<p><b>Resolution</b>: For <a href="/reference/dbt-jinja-functions/cross-database-macros">cross-database macros</a>, change <code>dbt_utils.MACRO_NAME()</code> to <code>dbt.MACRO_NAME()</code>.</p>
	</div>
</details>
<details>

> 	<summary><code>macro 'dbt_macro__generate_surrogate_key' takes not more than 1 argument(s)</code> </summary>

Reviewer: ```suggestion
	<summary>The macro <code>dbt_macro__generate_surrogate_key</code> takes not more than 1 argument(s) </summary>
```

Author: This is the exact error message and format, so can't change it

	<div>
		<p><b>Cause</b>: <code>generate_surrogate_key()</code> requires a single argument containing a list of columns, not a set of varargs.</p>
		<p><b>Resolution</b>: Change to <code>dbt_utils.generate_surrogate_key(['column_1', 'column_2'])</code> - note the square brackets. </p>
	</div>
</details>
<details>
	<summary>The <code>dbt_utils.surrogate_key</code> has been replaced by <code>dbt_utils.generate_surrogate_key</code></summary>
	<div>
		<p><b>Cause</b>: <code>surrogate_key()</code> has been replaced. </p>
		<p><b>Resolution</b>:
			<ol>
				<li>Decide whether you need to enable backwards compatibility <a href="#breaking-changes">as detailed above</a>.</li>
				<li>Find and replace <code>dbt_utils.surrogate_key</code> with <code>dbt_utils.generate_surrogate_key</code>.</li>
			</ol>
		</p>
	</div>
</details>
<details>

> 	<summary>macro <code>dbt_macro__test_expression_is_true</code> takes no keyword argument <code>condition</code></summary>

Reviewer: ```suggestion
	<summary>The macro <code>dbt_macro__test_expression_is_true</code> takes no keyword argument <code>condition</code></summary>
```

Author: Same here, but in doing this you have pointed out that there's inconsistent formatting of the different error messages in the migration guide (some are completely code bocks, some are a mix), so I will fix those!
